### PR TITLE
Add ShowAnswerXBlockMixin

### PR DESCRIPTION
**Pull request by noisecapella**
Part of https://github.com/mitodl/edx-sga/issues/185

Adds `ShowAnswerXBlockMixin` to provide a way to abstract the showanswer functionality from `CapaBase` into a class which can be referenced by xblocks like edx-sga

No manual testing yet (see related PR on edx-sga), just make sure tests pass.

### CODE REVIEW

File: xblockutils/show_answers.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        ]
+    )
+    solution = String(
+        help=_("Solution to the problem to show to the user"),
+        scope=Scope.settings,
+        multiline_editor='html',
+        resettable_editor=False,
+    )
+
+    def answer_available(self):  # pylint: disable=too-many-return-statements
```
Comment (noisecapella) [inline xblockutils/show_answers.py:45]: This is adapted from https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/capa_base.py#L892

File: xblockutils/show_answers.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+log = logging.getLogger(__name__)
+
+
+class ShowAnswerXBlockMixin(object):
+    """
+    Mixin for showanswer functionality
+    """
+
+    showanswer = String(
```
Comment (noisecapella) [inline xblockutils/show_answers.py:21]: Adapted from https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/capa_base.py#L133

File: tests/integration/test_show_answers.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+from xblock.core import XBlock
+from xblock.fields import Boolean, Dict, Float, Integer, List, String, DateTime
+from xblock.validation import ValidationMessage
+from xblockutils.base_test import SeleniumXBlockTest
+from tests.integration.utils import render_template
+from tests.integration.test_studio_editable import StudioEditableBaseTest
+from xblockutils.constants import ShowAnswer
+from xblockutils.show_answers import ShowAnswerXBlockMixin
+
+#from xblockutils.studio_editable_test import StudioEditableBaseTest, StudioContainerWithNestedXBlocksBaseTest
```
Comment (gsidebo) [inline tests/integration/test_show_answers.py]: Comment

File: tests/integration/test_show_answers.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        self.assertEqual(block.showanswer, ShowAnswer.PAST_DUE)
+
+    @XBlock.register_temp_plugin(ShowAnswerXBlock, "showanswer")
+    def test_fields(self):
+        """
+        Assert field overrides in XML
+        """
+        block = self.set_up_root_block('solution="The solution" showanswer="{}"'.format(ShowAnswer.ANSWERED))
+        self.assertEqual(block.solution, 'The solution')
+        self.assertEqual(block.showanswer, ShowAnswer.ANSWERED)
```
Comment (gsidebo) [inline tests/integration/test_show_answers.py:62]: ~~It doesn't look like anything in this class is being used. What's the reason for adding it in this PR?~~

EDIT: Sorry, disregard that. It's the `assert_unchanged` function that appears to be unused.

**Review by gsidebo — CHANGES_REQUESTED** [review]
A tiny comment and a question

File: tests/integration/test_show_answers.py (hunk 4 above)
Comment (noisecapella) [reply tests/integration/test_show_answers.py:62]: I copied it from `test_studio_editable`, I'll remove those functions

**Review by gsidebo — APPROVED** [review]
👍

**Pull request merged**